docs/understanding-dependencies.md

```diff
@@ -0,0 +1,24 @@
+# dependencies vs. devDependencies
```

Comment:
Nit: I think it'd be helpful to wrap `dependencies` and `devDependencies` in backticks to make them monospaced, to make it clear we're talking about a code thing.